querybook/server/tasks/run_datadoc.py

```diff
@@ -126,7 +131,21 @@ def _start_query_execution_task(
         datadoc_logic.append_query_executions_to_data_cell(
             cell_id, [query_execution_id], session=session
         )
-        return query_execution_id
+
+    # Run synchronously
+    query_run_status = run_query_task(
+        query_execution_id=query_execution_id,
+        execution_type=execution_type,
+        celery_task=self,
+    )
+    if (
+        retry["enabled"] is True
+        and retry["max_retries"] != 0
+        and query_run_status == QueryExecutionStatus.ERROR.value
+    ):
+        self.retry(countdown=retry["delay_sec"], max_retries=retry["max_retries"])
```

Comment:
this seems like it would infinite retry itself?

Reply:
Celery should take care of this. Testing shows that on cell run failure, the code retries up to the specified max_retries count. If running the cell continues to fail, and the code has retried the max_retries amount of times, then the code stops trying to run that cell.